testing/endpoint.go

```diff
@@ -790,6 +790,20 @@ func (endpoint *Endpoint) SetChannel(channel channeltypes.Channel) {
 	endpoint.Chain.App.GetIBCKeeper().ChannelKeeper.SetChannel(endpoint.Chain.GetContext(), endpoint.ChannelConfig.PortID, endpoint.ChannelID, channel)
 }
 
+// GetChannelUpgrade retrieves an IBC Channel Upgrade for the endpoint. The upgrade
+// is expected to exist otherwise testing will fail.
+func (endpoint *Endpoint) GetChannelUpgrade() channeltypes.Upgrade {
+	upgrade, found := endpoint.Chain.App.GetIBCKeeper().ChannelKeeper.GetUpgrade(endpoint.Chain.GetContext(), endpoint.ChannelConfig.PortID, endpoint.ChannelID)
+	require.True(endpoint.Chain.TB, found)
```

Comment:
I don't know if we should do this assertion in a getter. I would like to be able to use this function to assert an upgrade was not there yet. Or has been removed e.g. in an AbortUpgrade Test.

Maybe either a `GetChannelUpgrade() (channeltypes.Upgrade, found)`

or `GetChannelUpgrade() *channeltypes.Upgrade` where `nil` means not found. WDYT?

Reply:
Decided to go with this to align with the `path.Endpoint.GetChannel()` and `path.Endpoint.GetConnection()` methods. 

Personally I'd like to keep this one for happy path testing where you don't have to worry about checking the `found` boolean. Maybe for absence we can just do direct keeper invocation or add an additional helper?

Reply:
SGTM 👍